pkg/api/controller_test.go

```diff
@@ -5147,6 +5146,138 @@ func TestPeriodicTasks(t *testing.T) {
 	})
 }
 
+func TestSearchRoutes(t *testing.T) {
+	Convey("Upload image for test", t, func(c C) {
+		port := test.GetFreePort()
+		baseURL := test.GetBaseURL(port)
+		conf := config.New()
+		conf.HTTP.Port = port
+		tempDir := t.TempDir()
+
+		ctlr := api.NewController(conf)
+		ctlr.Config.Storage.RootDirectory = tempDir
+
+		go startServer(ctlr)
+		defer stopServer(ctlr)
+
+		test.WaitTillServerReady(baseURL)
+
+		repoName := "testrepo"
+		cfg, layers, manifest, err := test.GetImageComponents(10000)
+		So(err, ShouldBeNil)
+
+		err = test.UploadImage(
+			test.Image{
+				Config:   cfg,
+				Layers:   layers,
+				Manifest: manifest,
+				Tag:      "latest",
+			}, baseURL, repoName)
+
+		So(err, ShouldBeNil)
+
+		Convey("GlobalSearch with authz enabled", func(c C) {
+			conf := config.New()
+			port := test.GetFreePort()
+			baseURL := test.GetBaseURL(port)
+
+			user1 := "test"
+			password1 := "test"
+			testString1 := getCredString(user1, password1)
+			htpasswdPath := test.MakeHtpasswdFileFromString(testString1)
+			defer os.Remove(htpasswdPath)
+			conf.HTTP.Auth = &config.AuthConfig{
+				HTPasswd: config.AuthHTPasswd{
+					Path: htpasswdPath,
+				},
+			}
+
+			conf.HTTP.Port = port
+
+			defaultVal := true
+
+			searchConfig := &extconf.SearchConfig{
+				Enable: &defaultVal,
+			}
+
+			conf.Extensions = &extconf.ExtensionConfig{
+				Search: searchConfig,
+			}
+
+			conf.AccessControl = &config.AccessControlConfig{
+				Repositories: config.Repositories{
+					repoName: config.PolicyGroup{
+						Policies: []config.Policy{
+							{
+								Users:   []string{user1},
+								Actions: []string{"read"},
+							},
+						},
+						DefaultPolicy: []string{},
+					},
+				},
+				AdminPolicy: config.Policy{
+					Users:   []string{},
+					Actions: []string{},
+				},
+			}
+
+			ctlr := api.NewController(conf)
+
+			ctlr.Config.Storage.RootDirectory = tempDir
+
+			go startServer(ctlr)
+			defer stopServer(ctlr)
+			test.WaitTillServerReady(baseURL)
+
+			query := `
+			{
+				GlobalSearch(query:""){
+					Repos {
```

Comment:
I think we should have a test according to which we verify that one repo is returned because the user has permissions to view it, while another is not returned as the user doesn't have permissions to view the second one.